view/adminhtml/web/js/components/query-merchandising.js

```diff
@@ -26,12 +25,12 @@ define(
 					$( document ).ready(function() {
 					    self.initSubscribers();
 					});
-					
+
 				},
 
 				initObservable: function () {
 					this._super().observe(
-						'queryText storeId'
```

Comment:
I am curious why we sometimes handle events with Knockout observables and  sometimes with jQuery?

Reply:
From what I remember, Query Merchandiser is different from the LPB because there are no collapsible sections in it so managing the events was possible from knockout observables and not for the LPB (because the merchandising section is not deployed by default). 

I had to remove it in this PR because the knockout observable automatically triggered the InstantSearch refresh **before** it was re-initialized with the right application so there was a weird behaviour where 2 calls were made (one with the "old" application (knockout observable) and one from the "new" one (jQuery)), which led to a JS error because the index wasn't found on the "old" one. 

This change makes sure that the InstantSearch refresh is performed by jQuery on the right application and doesn't generate any error.

Reply:
Ok that makes sense. Thanks for the added context. Seems like we should just stick with one consistent approach for "grokability" and since jQuery seems to cover both use cases, this looks like a move in the right direction. 🤘